### PR TITLE
OpenAI-DotNet 6.5.4

### PR DESCRIPTION
**Pull request by johnkors**
- Added [`response_format`](https://platform.openai.com/docs/api-reference/images/create#images/create-response_format) to `ImageGenerationRequest`

### CODE REVIEW

File: OpenAI-DotNet/Images/ImageGenerationRequest.cs [hunk 1]
```diff
@@ -66,5 +66,11 @@ public ImageGenerationRequest(string prompt, int numberOfResults = 1, ImageSize
         /// </summary>
         [JsonPropertyName("user")]
         public string User { get; }
+
+        /// <summary>
+        /// Optional: The format in which the generated images are returned. Must be one of url or b64_json. Defaults to url
+        /// </summary>
+        [JsonPropertyName("response_format")]
+        public string ResponseFormat { get; } = "url;
```
Comment (StephenHodgson) [inline OpenAI-DotNet/Images/ImageGenerationRequest.cs:74]:
```suggestion
        public string ResponseFormat { get; } = "url";
```

**Review by StephenHodgson — CHANGES_REQUESTED** [review]
Thanks for the contribution @johnkors!

You also need to bump the version and add change notes

**Comment by StephenHodgson** [comment]
@johnkors I think you also need to update the endpoint to properly send the data as well.

**Comment by johnkors** [comment]
> You also need to bump the version and add change notes

Ah, you're doing it manually. Got it. Fixed now.

Considered [automating the versioning](https://github.com/johnkors/dotnet-library/blob/main/.github/workflows/Release.yml#L29), btw..? Pretty easy to setup.

**Comment by StephenHodgson** [comment]
> Considered [automating the versioning](https://github.com/johnkors/dotnet-library/blob/main/.github/workflows/Release.yml#L29), btw..? Pretty easy to setup.

I have but I want all the historical notes so far and sometimes GitHub doesn't generate the release notes properly. It's just been easier for me to copy.

File: OpenAI-DotNet/Images/ImageGenerationRequest.cs [hunk 2]
```diff
@@ -66,5 +66,11 @@ public ImageGenerationRequest(string prompt, int numberOfResults = 1, ImageSize
         /// </summary>
         [JsonPropertyName("user")]
         public string User { get; }
+
+        /// <summary>
+        /// Optional: The format in which the generated images are returned. Must be one of url or b64_json. Defaults to url
```
Comment (StephenHodgson) [inline OpenAI-DotNet/Images/ImageGenerationRequest.cs:71]: So the ImageResult.cs expects a url. we need to update the endpoint to handle the new deserialized format.

**Review by StephenHodgson — CHANGES_REQUESTED** [review]

**Comment by johnkors** [comment]
> @johnkors I think you also need to update the endpoint to properly send the data as well.

Ah, right - next time I'll not do this directly from the GitHub UI :D Sorry for the mess.

File: OpenAI-DotNet/Images/ImageGenerationRequest.cs (hunk 2 above)
Comment (johnkors) [reply OpenAI-DotNet/Images/ImageGenerationRequest.cs]: Ok, then I guess it's a breaking change (major bump)? I was using version 6.3.1 and it was returning a string. Was there a breaking change on a minor bump making it an URI?

Comment (StephenHodgson) [reply OpenAI-DotNet/Images/ImageGenerationRequest.cs]: Nah, I don't think it's a breaking change if you do it right.

Comment (StephenHodgson) [reply OpenAI-DotNet/Images/ImageGenerationRequest.cs]: > Was there a breaking change on a minor bump making it an URI?

I don't understand the question

Comment (StephenHodgson) [reply OpenAI-DotNet/Images/ImageGenerationRequest.cs]: > Ok, then I guess it's a breaking change (major bump)?

This shouldn't be a major or minor if you do it right.

**Comment by StephenHodgson** [comment]
Was there any particular reason why you needed base64_json?

File: OpenAI-DotNet/Images/ImageGenerationRequest.cs [hunk 3]
```diff
@@ -66,5 +68,11 @@ public ImageGenerationRequest(string prompt, int numberOfResults = 1, ImageSize
         /// </summary>
         [JsonPropertyName("user")]
         public string User { get; }
+
+        /// <summary>
+        /// Optional: The format in which the generated images are returned. Must be one of url or b64_json. Defaults to "url"
```
Comment (StephenHodgson) [inline OpenAI-DotNet/Images/ImageGenerationRequest.cs:73]:
```suggestion
        /// The format in which the generated images are returned. Must be one of url or b64_json. Defaults to "url"
```

File: OpenAI-DotNet/Images/ImageGenerationRequest.cs (hunk 2 above)
Comment (johnkors) [reply OpenAI-DotNet/Images/ImageGenerationRequest.cs]: Ah, right. Agreed. The API returns a _different_ property. The docs did not mention it, but I just tested it now. I'll add the new property `b64_json` to ImageResult.

**Comment by johnkors** [comment]
> Was there any particular reason why you needed base64_json?

The URLs returned from OpenAI are very short lived. They include a short lived access to Azure Blob Storage. 1 hour or so. Wanted to see if I could just pipe base64 instead of having my own storage.

**Comment by johnkors** [comment]
Got a simple POC up. Here's what I'm building btw (piping OpenAI images to Slack).

image
image

**Comment by StephenHodgson** [comment]
I see, so the image is actually encoded in the response data

**Comment by StephenHodgson** [comment]
So this PR is good to go for you @johnkors ?

**Comment by johnkors** [comment]
Yup!

**Comment by StephenHodgson** [comment]
idk, I think we still need to add the new parameter to the form in the ImageEndpoint class.

**Comment by johnkors** [comment]
> idk, I think we still need to add the new parameter to the form in the ImageEndpoint class.

I think I understood what you meant. Updated now.

**Comment by StephenHodgson** [comment]
@johnkors I think we are sill missing the form updates in the [ImagesEndpoint](https://github.com/RageAgainstThePixel/OpenAI-DotNet/blob/40c3fc2208e488dc52d82bdb22d751bfee673c6b/OpenAI-DotNet/Images/ImagesEndpoint.cs#L103)

**Comment by johnkors** [comment]
> @johnkors I think we are sill missing the form updates in the [ImagesEndpoint](https://github.com/RageAgainstThePixel/OpenAI-DotNet/blob/40c3fc2208e488dc52d82bdb22d751bfee673c6b/OpenAI-DotNet/Images/ImagesEndpoint.cs#L103)

I kept other endpoints out of scope for now, but sure – it can be included.

**Comment by johnkors** [comment]
Added `response_format` to `images/variations` as well, since that was missing as well.

**Comment by johnkors** [comment]
Ugh, I see now that the main project where you've forked this from already has support for this. Not sure why there needs to be 2 C# wrapper SDKs for a single HTTP API. I'll probably migrate to that one. I'll leave this PR open if you want to include, though.

Good luck.

**Comment by StephenHodgson** [comment]
If that's what you really want to do 😅
But I keep this library up to date a lot more than he does. 
I also feel like it's a lot more complete and cleaner.

Any of the case thanks for bringing this up and doing the work. I'll try to get this merged in today.

**Comment by johnkors** [comment]
>But I keep this library up to date a lot more than he does.
> I also feel like it's a lot more complete and cleaner.

Fair point!

**Pull request merged**